Source/WebCore/editing/cocoa/HTMLConverter.mm

```diff
@@ -2613,8 +2640,21 @@ AttributedString editingAttributedString(const SimpleRange& range, OptionSet<Inc
         else
             text = makeStringByReplacingAll(it.text(), noBreakSpace, ' ');
 
+        auto linkURL = [&] -> URL {
+            RefPtr enclosingLink = enclosingLinkElement(*node, enclosingLinkCache);
+            if (!enclosingLink)
+                return { };
+
+            return enclosingLink->absoluteLinkURL();
+        }();
+
+        if (linkURL.isEmpty())
+            [attributes removeObjectForKey:NSLinkAttributeName];
+        else
+            [attributes setObject:(NSURL *)linkURL forKey:NSLinkAttributeName];
```

Comment:
maybe this would be better off going in `updateAttributesForStyle`? since that’s where all the other attributes that have the pattern of removeObjectForKey / setObject are in, though doesn’t really matter much

Reply:
We could — but we should then rename `updateAttributesForStyle`, since this link attribute doesn't depend on style at all (just DOM attributes and DOM structure). I'll rename it to `updateAttributes` and pull the logic into there.

Thanks for the review!